packages/compass/src/main/telemetry.ts

```diff
@@ -126,50 +126,43 @@ class CompassTelemetry {
   }
 
   private static _init(app: typeof CompassApplication) {
-    process.on('compass:track', (meta: EventInfo) => {
-      this._track(meta);
-    });
-
-    ipcMain.respondTo('compass:track', (evt, meta: EventInfo) => {
-      (process as EventEmitter).emit('compass:track', meta);
-    });
-
-    ipcMain.respondTo(
-      'compass:usage:identify',
-      (evt, { currentUserId, telemetryAnonymousId }: { currentUserId?: string; telemetryAnonymousId: string }) => {
-        // This always happens after the first enable/disable call.
-        this.currentUserId = currentUserId;
-        this.telemetryAnonymousId = telemetryAnonymousId;
-        void this.identify();
-      }
-    );
-
-    ipcMain.respondTo('compass:usage:enabled', () => {
-      log.info(
-        mongoLogId(1_001_000_094),
-        'Telemetry',
-        'Enabling Telemetry reporting'
-      );
-      if (this.state !== 'enabled') {
+    const { trackUsageStatistics, currentUserId, telemetryAnonymousId } = preferences.getPreferences();
+    this.currentUserId = currentUserId;
+    this.telemetryAnonymousId = telemetryAnonymousId;
+    if (trackUsageStatistics) {
+      this.state = 'enabled';
+      void this.identify();
+    }
+    preferences.onPreferenceValueChanged('trackUsageStatistics', value => {
```

Comment:
If we are listening to the preferences changes here now, does this mean that we can remove this check from compass-logging?

Reply:
Talked f2f: I completely forgot that another reason we are doing a check in logging is so that we can completely skip running "pricey" telemetry events that require getting info from the server